src/meshapi/tests/test_update_panos_github.py

```diff
@@ -0,0 +1,136 @@
+import os
+from django.test import Client, TestCase
+from django.contrib.auth.models import User
+
+from meshapi.models import Building, Install, Link, Member, Sector
+from meshapi.views import panoramas
+from .sample_data import sample_building, sample_install, sample_member
+
+
+class TestPanoPipeline(TestCase):
+    c = Client()
+
+    def setUp(self):
+        sample_install_copy = sample_install.copy()
+        self.building_1 = Building(**sample_building)
+        self.building_1.save()
+        sample_install_copy["building"] = self.building_1
+
+        self.member = Member(**sample_member)
+        self.member.save()
+        sample_install_copy["member"] = self.member
+
+        self.install = Install(**sample_install_copy)
+        self.install.save()
+
+        self.admin_user = User.objects.create_superuser(
+            username="admin", password="admin_password", email="admin@example.com"
+        )
+        self.c.login(username="admin", password="admin_password")
+
+    def test_set_panoramas(self):
+        # Fabricate some fake panorama photos
+        n = self.install.install_number
+        panos = {n: [f"{n}.jpg", f"{n}a.jpg"]}
+
+        panoramas.set_panoramas(panos)
+
+        # Now check that that worked.
+        building = Building.objects.get(id=self.install.building.id)
+        saved_panoramas = [
+            f"https://node-db.netlify.app/panoramas/{n}.jpg",
+            f"https://node-db.netlify.app/panoramas/{n}a.jpg",
+        ]
+        self.assertEqual(saved_panoramas, building.panoramas)
+
+
+class TestPanoAuthentication(TestCase):
+    c = Client()
+    admin_c = Client()
+
+    def setUp(self):
+        self.admin_user = User.objects.create_superuser(
+            username="admin", password="admin_password", email="admin@example.com"
+        )
+        self.admin_c.login(username="admin", password="admin_password")
+
+    def test_update_panoramas_unauthenticated(self):
+        response = self.c.get("/api/v1/update-panoramas/")
+        code = 403
+        self.assertEqual(
+            code,
+            response.status_code,
+            f"status code incorrect. Should be {code}, but got {response.status_code}",
+        )
+
+    # This tests the endpoint, but not the actual full pipeline.
```

Comment:
Why doesn't this test the full pipeline?

Reply:
I'm lazy, and when I wrote that test I hadn't yet broken the code up to be easily unit tested yet. I think this is now a relatively atomic test. Testing the whole thing front-to-back is probably unnecessary now.